solutions/python-igraph/mean_degree_neighborhood/mean_degree_neighborhood.py

```diff
@@ -0,0 +1,20 @@
+import igraph as ig
+
+def mean_neighborhood_degree(g: ig.Graph, k: int):
+    n = len(g.vs)
+    res = [0] * n
+    for v in g.vs:
+        kneighborhood = []
+        distances = g.distances(v, g.vs, weights=None, mode='all')[0]
+        for i in range(n):
+            if distances[i] == k:
```

Comment:
I think this should be `<= k` here, not `== k`, but maybe @szhorvat meant something different when he originally specified the benchmark.

Nevertheless, you can simplify this greatly with a Python list comprehension. Also note that you need to exclude `v` itself:

```
kneighborhood = [i for i, d in enumerate(distances) if d <= k and i != v.index]
```

Reply:
Yes, I meant `<= k`.

Reply:
Regarding `distances = g.distances(v, g.vs, weights=None, mode='all')[0]`,

 - `target` should be left at `None`. @ntamas In the C layer, the `igraph_vss_all()` path is better optimized for many functions, and will perform better than listing all vertices explicitly. I assume `None` maps to `igraph_vss_all()`. What about `g.vs`?  Does it map to `vss_all()` or to something for which `igraph_vs_is_all()` is not `true`?
 - Generally, computing the distance to all other vertices, even those further than `k` steps away, will be quite inefficient. Using the `neighborhood()` method will be better here.